lib/demuxer.ex

```diff
@@ -0,0 +1,161 @@
+defmodule Membrane.Ogg.Demuxer do
+  use Membrane.Filter
+  alias Membrane.{Buffer, RemoteStream, Opus}
+
+  def_input_pad :input,
+    availability: :always,
+    mode: :pull,
+    demand_unit: :buffers,
+    accepted_format: any_of(RemoteStream)
+
+  def_output_pad :output,
+    availability: :on_request,
+    mode: :pull,
+    accepted_format: %RemoteStream{type: :packetized, content_format: Opus}
+
+  defmodule State do
+    @type t :: %__MODULE__{
+            actions_buffer: list,
+            parser_acc: binary,
+            phase: :all_outputs_linked | :awaiting_linking,
+            continued_packets: %{}
+          }
+
+    defstruct actions_buffer: [],
+              parser_acc: <<>>,
+              phase: :awaiting_linking,
+              continued_packets: %{}
+  end
+
+  @impl true
+  def handle_init(_ctx, _options) do
+    {[], %State{}}
+  end
+
+  @impl true
+  def handle_playing(_context, state) do
+    {[{:demand, :input}], state}
+  end
+
+  @impl true
+  def handle_pad_added(Pad.ref(:output, id), _context, state) do
+    {[
+       stream_format:
+         {Pad.ref(:output, id), %RemoteStream{type: :packetized, content_format: Opus}}
+     ], %State{state | phase: :all_outputs_linked}}
+  end
+
+  @impl true
+  def handle_process(:input, %Buffer{payload: bytes}, context, state) do
+    unparsed = state.parser_acc <> bytes
+
+    {parsed, unparsed, continued_packets} =
+      Membrane.Ogg.Parser.parse(unparsed, state.continued_packets)
+
+    state = %State{state | parser_acc: unparsed}
+    state = %State{state | continued_packets: continued_packets}
+
+    {actions, state} = process_packets({[], context, state}, parsed)
+
+    demand_if_not_blocked({actions, state})
+  end
+
+  @impl true
+  def handle_demand(Pad.ref(:output, _id), _size, :buffers, context, state)
+      when state.phase == :all_outputs_linked do
+    {[], state}
+    |> reclassify_buffer_actions(context)
+    |> demand_if_not_blocked()
+  end
+
+  @impl true
+  def handle_demand(Pad.ref(:output, _id), _size, :buffers, _context, state) do
+    {:ok, state}
+  end
+
+  defp process_packets({actions, context, state}, packets_list) do
```

Comment:
You don't need to pass `actions` here. I would also flatten arguments passed to the function: 

```suggestion
  defp process_packets(packets, context, state) do
```